### DISCUSSION TITLE
How to not output some variables in output file

### DISCUSSION THREAD

**Discussion by pshen20127** [idaholab/moose#25291] [Q&A Modules: Solid mechanics]
Hello All

After I submit a job, if it converge at that timestep, it will output the values of variables at the current timestep in output file like below. Since outputing them will take much time, I want to turn it off. Do you know how to do that?
![Screenshot from 2023-08-23 16-57-35](https://github.com/idaholab/moose/assets/81499506/e50749f1-99cc-41ed-8e20-e47451535c4a)

Pshen

**Comment by GiudGiud** [comment]
Hello

You can specify an interval for output, for all types of output. From the console, to exodus to CSV

and you can turn them on/off using the execute_on flag too (execute_on = FINAL only runs at the end for example)

Guillaume

**Reply by echo1115** [reply]
The parameter `show` or `hide` could control the variables and postprocessors that should or shoulde NOT be output. For example, in `test/tests/auxkernels/divergence_aux/test_fv.i`  [https://github.com/idaholab/moose/blob/42c6f190d9ea2a13de6f00604e582531b6f580dc/test/tests/auxkernels/divergence_aux/test_fv.i#L156](url)

**Reply by pshen20127** [reply]
@GiudGiud Which block in input file should I add in 'execute_on = FINAL'?

@echo1115 Thank you. I tried your suggestion but it doesn't work. It still output variables.

**Reply by GiudGiud** [reply]
in your outputs block you can pass the execute_on / hide/ show parameters

```
[Outputs]
  
  [out]
    type = Exodus
    hide = 'this vairable'
    execute_on = FINAL. OR interval = 100
 []
[]
```

**Reply by pshen20127** [reply]
Thank you. the command 'interval' works. But don't know why 'execute_on = FINAL' and 'hide' don't work.

**Reply by GiudGiud** [reply]
Should both work or error. 
If you make a minimal non-working MOOSE-only example I can take a look at it

**Reply by pshen20127** [reply]
I put test files in the link below. Please take a look.
https://github.com/pshen20127/Test.git

**Reply by GiudGiud** [reply]
there s no interval / particular output parameters there. which variable did you try to hide?

**Reply by echo1115** [reply]
the `hide` should work. For example, when running case in `test/tests/auxkernels/divergence_aux/test_fv.i`, it shows:

![output1](https://github.com/idaholab/moose/assets/69078335/841ab619-eb73-42cf-bdaa-c4eb09053226)

and if you comment the `hide` command:
```
[Outputs]
  csv = true
  #hide = 's1 s2 s3 s4'
[]
```
The 's1 s2 s3 s4' will appear in the results:

![output2](https://github.com/idaholab/moose/assets/69078335/883f01b1-ffad-4682-ab95-c6234353c45a)

@pshen20127 Maybe you can show us how you use the `hide `command in input file.

**Reply by pshen20127** [reply]
@GiudGiud Sorry, I have updated input file in outputs block as below. You can try runing job with  `hide` or `execute_on = FINAL`. 
```
[Outputs]
  file_base = test
  exodus = true
  csv=true
  hide = 'euler_angle_2'
  #execute_on = FINAL
[]
```

@echo1115 Yes, it works when I run `test/tests/auxkernels/divergence_aux/test_fv.i`. But I don't know why it doesn't work in my case. I used `hide` command as above. But still get 'euler_angle_2' output. You can try files in the link: https://github.com/pshen20127/Test.git
![Screenshot from 2023-08-25 11-25-29](https://github.com/idaholab/moose/assets/81499506/040589ed-644d-42e0-bfb1-047db132c2da)

**Reply by ykvishal** [reply]
You could try this [show_hide.i](https://github.com/idaholab/moose/blob/next/test/tests/outputs/postprocessor/show_hide.i)

create a separate block for csv in the output
```
  [./out]
    type = CSV
    show = 'elem_56'
    hide = 'elem_12'
  [../]
```

**Reply by GiudGiud** [reply]
yeah or for the console:
  [console]
    type = Console
    hide = 'euler_angle_2'
  [../]
